sqlserver/datadog_checks/sqlserver/database_metrics/database_agent_metrics.py

```diff
@@ -0,0 +1,159 @@
+from datadog_checks.base.config import is_affirmative
+
+from datadog_checks.sqlserver.const import  AWS_RDS_HOSTNAME_SUFFIX
+
+from .base import SqlserverDatabaseMetricsBase
+
+AGENT_ACTIVITY_DURATION_QUERY = {
+    "name": "msdb.dbo.sysjobactivity",
+    "query": """\
+        SELECT
+            sj.name,
+            ja.job_id,
+            DATEDIFF(SECOND, ja.start_execution_date, GETDATE()) AS duration_seconds
+        FROM msdb.dbo.sysjobactivity AS ja
+        INNER JOIN msdb.dbo.sysjobs AS sj
+        ON ja.job_id = sj.job_id
+        WHERE ja.start_execution_date IS NOT NULL
+            AND ja.stop_execution_date IS NULL
+            AND session_id = (
+                SELECT MAX(session_id) 
+                FROM msdb.dbo.sysjobactivity
+            )
+    """,
+    "columns": [
+        {"name": "job_name", "type": "tag"},
+        {"name": "job_id", "type": "tag"},
+        {"name": "agent.active_jobs.duration", "type": "gauge"},
+    ],
+}
+
+AGENT_ACTIVITY_STEPS_QUERY = {
+    "name": "msdb.dbo.sysjobactivity",
+    "query": """\
+        WITH ActiveJobs AS (
+            SELECT
+                ja.job_id,
+                last_executed_step_id
+            FROM msdb.dbo.sysjobactivity AS ja
+            WHERE ja.start_execution_date IS NOT NULL
+                AND ja.stop_execution_date IS NULL
+                AND session_id = (
+                    SELECT MAX(session_id) 
+                    FROM msdb.dbo.sysjobactivity
+                )
+        ),
+        CompletedSteps AS (
+            SELECT
+                sjh1.job_id,
+                sjh1.step_id,
+                sjh1.step_name,
+                sjh1.run_status
+            FROM msdb.dbo.sysjobhistory AS sjh1
+            WHERE NOT EXISTS (
+                SELECT 1
+                FROM msdb.dbo.sysjobhistory AS sjh2
+                WHERE sjh2.job_id = sjh1.job_id
+                    AND sjh2.step_id = 0
+                    AND sjh2.instance_id > sjh1.instance_id
+                )
+        )
+        SELECT
+            j.name,
+            aj.job_id,
+            cs.step_name,
+            cs.step_id,
+            cs.run_status AS step_run_status,
+            1 AS step_info
+        FROM ActiveJobs AS aj
+        INNER JOIN CompletedSteps AS cs
+        ON aj.job_id = cs.job_id
+            AND aj.last_executed_step_id = cs.step_id
+        INNER JOIN msdb.dbo.sysjobs AS j
+        ON j.job_id = aj.job_id
+    """,
+    "columns": [
+        {"name": "job_name", "type": "tag"},
+        {"name": "job_id", "type": "tag"},
+        {"name": "step_name", "type": "tag"},
+        {"name": "step_id", "type": "tag"},
+        {"name": "step_run_status", "type": "tag"},
+        {"name": "agent.active_jobs.step_info", "type": "gauge"},
+
+    ],
+}
+
+AGENT_ACTIVE_SESSION_DURATION_QUERY = {
```

Comment:
Why are we reporting active session duration? How can user use `session_id` in the job context?

Reply:
the idea is to create a metric for active job duration tagged by job_name and id IIRC

Reply:
The query under `AGENT_ACTIVE_SESSION_DURATION_QUERY` doesn't have `job_name` tags: https://github.com/DataDog/integrations-core/blob/0bbd2da6cf90ebb8b5e8d3ea241eb677f1633fdf/sqlserver/datadog_checks/sqlserver/database_metrics/database_agent_metrics.py#L96-L97

Reply:
There was a request in the RFC regarding the health of the agent itself. This query gives information about how long the active (most recent) session of the sqlserver agent itself has been running. The available data about the session is rather sparse https://learn.microsoft.com/en-us/sql/relational-databases/system-tables/dbo-syssessions-transact-sql?view=sql-server-ver16